src/plugins/discover_shared/public/services/discover_features/types.ts

```diff
@@ -30,8 +32,39 @@ export interface ObservabilityLogsAIAssistantFeature {
   render: (deps: ObservabilityLogsAIAssistantFeatureRenderDeps) => JSX.Element;
 }
 
+/** ***************************** Security **********/
+export interface SecuritySolutionCellRenderFeature {
+  id: 'security-solution-cell-render';
+  getRender: () => Promise<
+    (fieldName: string) => FunctionComponent<DataGridCellValueElementProps> | undefined
+  >;
+}
+
+export interface SecuritySolutionAppWrapperFeatureArgs {
+  store: unknown;
+}
+
+export interface SecuritySolutionAppWrapperFeature {
+  id: 'security-solution-app-wrapper';
+  getWrapper: () => Promise<
+    (args: SecuritySolutionAppWrapperFeatureArgs) => FunctionComponent<PropsWithChildren<{}>>
+  >;
+}
+
+export interface SecuritySolutionReduxStoreInitFeature {
+  id: 'security-solution-redux-store-init';
+  /* returns the security's redux store instance */
+  get: () => Promise<unknown>;
```

Comment:
Ideally we'd have some type for this if possible, even if we can't type it fully.

Reply:
The problem is `SecurityAppStore` type is constructed inside security solution using `types` from multiple subplugins as shown below. To be more precise, I can make it `object` but other than that, for now, it is not trival to extract these types.

We anyway do not plan to do any kind of `store` manipulation inside discover.

https://github.com/elastic/kibana/blob/7538565b6db8f695672cdbd4c96185e6e9b26d19/x-pack/plugins/security_solution/public/common/store/types.ts#L30-L47